weldx/asdf/validators.py

```diff
@@ -95,6 +95,261 @@ def _unit_validator(
         )
 
 
+def _compare(_int, exp_string):
+    """Compare helper of two strings for _custom_shape_validator.
+
+
+    Return a boolean.
+
+    Parameters
+    ----------
+    _int:
+        Integer
+    exp_string:
+        String with the expected dimension
+
+    Returns
+    -------
+    bool
+        True or False
+    """
+    if _int < 0:
+        raise ValueError("Negative dimension found")
+    if ":" in exp_string:
+        ranges = exp_string.split(":")
+        if ranges[0] == "" and ranges[1] == "":
+            return True
+        else:
+            if ranges[0] != "" and ranges[1] != "":
+                if int(ranges[0]) <= int(ranges[1]):
+                    pass
+                else:
+                    raise ValueError(
+                        f"The range should not be descending in {exp_string}"
+                    )
```

Comment:
using a `>` instead of `<=` should get rid of the unused branch (`pass`). Then you can also combine it with the outer branch. So this should work:

~~~ python
if ranges[0] != "" and ranges[1] != "" and int(ranges[0]) > int(ranges[1]):
    raise ValueError(f"The range should not be descending in {exp_string}")
~~~